stdlib/email/src/main/java/org/ballerinalang/stdlib/email/util/SmtpUtil.java

```diff
@@ -74,6 +75,15 @@ public static Properties getProperties(MapValue smtpConfig, String host) {
         properties.put(EmailConstants.PROPS_SMTP_AUTH, "true");
         properties.put(EmailConstants.PROPS_SMTP_STARTTLS, "true");
         properties.put(EmailConstants.PROPS_ENABLE_SSL, smtpConfig.getBooleanValue(EmailConstants.PROPS_SSL));
+        CommonUtil.addCustomProperties(smtpConfig.getArrayValue(EmailConstants.PROPS_PROPERTIES), properties);
+        if (log.isDebugEnabled()) {
+            Set<String> propertySet = properties.stringPropertyNames();
+            log.debug("SMTP Properties set are as follows.");
+            for (Object propertyObj : propertySet) {
+                log.debug("Property Name: " + propertyObj + ", Value: " + properties.get(propertyObj).toString()
+                        + " ValueType: " + properties.get(propertyObj).getClass().getName());
+            }
+        }
```

Comment:
I think we need to give priority to the predefined fields? shall we first add custom properties and then add field values.

Reply:
Custom properties are given higher priority to override the default functionality of the connector for advanced users.